includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -826,6 +827,12 @@ public function payment_fields() {
 			?>
 
 			</fieldset>
+
+
+			<?php if ( Fraud_Prevention_Service::instance()->is_enabled() ) : ?>
+				<input type="hidden" id="wcpay-fraud-prevention-token" name="wcpay-fraud-prevention-token" value="<?php echo esc_attr( Fraud_Prevention_Service::instance()->get_token() ); ?>">
```

Comment:
W have a constant for value in `name` attribute.

Reply:
Similarly to https://github.com/Automattic/woocommerce-payments/pull/3979/files#r832202589 I feel that having to use `Fraud_Prevention_Service` twice to create the input indicates that whole input creation should the the responsibility of `Fraud_Prevention_Service` with a method `get_hidden_input`.

Reply:
I see. Makes sense, let's stick to the current approach then.

Reply:
I'm not sure about it. I agree with you that the name of field should be a constant on `Fraud_Prevention_Service` so that we are using the same value when creating the input and pulling it out from `$_POST`. 

What I don't like is having to use same class twice in the same line which is what's happening for cases. That's why I feel that `Fraud_Prevention_Service` should create the whole input as well as verify token using whole `$_POST`  because the the constant is completely isolated into the service.

Reply:
I understood, the key point here is the separation of concerns. This service has enough responsibilities, adding html-generation is not teething we should there.

That's why my last comment was to use the string and not constant.